File: dynamic_listings/lua/default/computer_science/head_recursion.lua
local function count_forwards(n)
    -- Stop condition
    if (n == 0) then
        -- If we don't do this, we won't print 0
        print(n)
        return
    end
    -- Recursive call
    count_forwards(n-1)
    -- Procedure
    print(n)
end

File: dynamic_listings/rust/default/computer_science/head_recursion.rs
fn count_forwards(n: u32) {
    // Stop condition
    if n == 0 {
        // If we don't do this, we won't print 0
        println!("{n}");
        return;
    }
    // Recursive call
    count_forwards(n - 1);
    // Procedure
    println!("{n}");
}
